File: topics/examples/redis-stream-producer-example.py
#!/usr/bin/env python3
"""
Redis Stream Producer示例
演示如何使用redis-py库发送消息到Redis Stream
"""

import redis
import json
import time

# Redis连接配置
REDIS_HOST = 'localhost'
REDIS_PORT = 6379
REDIS_DB = 0

def create_redis_client():
    """创建Redis客户端"""
    return redis.Redis(
        host=REDIS_HOST,
        port=REDIS_PORT,
        db=REDIS_DB,
        decode_responses=True
    )

def send_message():
    """发送消息到Redis Stream"""
    client = create_redis_client()
    stream_name = 'events:orders'
    
    for i in range(10):
        message = {
            'order_id': f'ORD-{i:04d}',
            'user_id': f'USER-{i}',
            'amount': 100.0 + i,
            'timestamp': time.time()
        }
        
        # 添加消息到Stream
        message_id = client.xadd(
            stream_name,
            message,
            maxlen=10000  # 最大长度10000
        )
        print(f"Sent: {message_id} - {message['order_id']}")
    
    print("Messages sent successfully")

if __name__ == '__main__':
    send_message()

File: topics/examples/redis-stream-producer-example.java
package com.example.mq;

import redis.clients.jedis.Jedis;
import redis.clients.jedis.StreamEntryID;
import java.util.HashMap;
import java.util.Map;

public class RedisStreamProducer {
    private static final String STREAM_NAME = "events:orders";
    
    public static void main(String[] args) {
        // 连接Redis
        Jedis jedis = new Jedis("localhost", 6379);
        
        try {
            // 发送消息
            for (int i = 0; i < 10; i++) {
                Map<String, String> fields = new HashMap<>();
                fields.put("order_id", String.format("ORD-%04d", i));
                fields.put("user_id", "USER-" + i);
                fields.put("amount", String.valueOf(100.0 + i));
                fields.put("timestamp", String.valueOf(System.currentTimeMillis()));
                
                // 添加消息到Stream
                StreamEntryID messageId = jedis.xadd(
                    STREAM_NAME,
                    StreamEntryID.NEW_ENTRY,
                    fields
                );
                
                System.out.println("Sent: " + messageId + " - ORD-" + String.format("%04d", i));
            }
            
            System.out.println("Messages sent successfully");
        } finally {
            jedis.close();
        }
    }
}
